### ISSUE TITLE
Patch S1438 ('semi') : Report on proper statement end line

### ISSUE THREAD

**Issue by yassin-kammoun-sonarsource** [SonarSource/SonarJS#3127]
ESLint (['semi'](https://eslint.org/docs/rules/semi)) rule, known also as S1438, currently flags statements missing semicolons on the wrong line end on some occasions. This produces misleading issue highlightings on SonarQube (see [example](https://peach.sonarsource.com/project/issues?id=goof_nodev14&issues=AX4HMu-JcYOUBzznwtq_&open=AX4HMu-JcYOUBzznwtq_)).

Consider the following snippet:

```javascript
/* 1 */ function f() {
/* 2 */    if (cond()) {
/* 3 */        return foo() // missing semicolon
/* 4 */    } else {
/* 5 */        return bar();
/* 6 */    }
/* 7 */ }
```

ESLint reports an issue for the missing semicolon of `return foo()` with _line start 3_, _line end 4_, and _column end 0_. We either need to patch ESLint issue locations for this rule with a decorator or suggest a change in ESLint's implementation.

Reported in https://community.sonarsource.com/t/missing-semicolon-in-js/61998
